cirras-underwriting-api/cirras-underwriting-api-rest-endpoints/src/test/java/ca/bc/gov/mal/cirras/underwriting/api/rest/v1/endpoints/ProductEndpointTest.java

```diff
@@ -0,0 +1,371 @@
+		fetchedResource.setDeductibleLevel(30);
+		fetchedResource.setInsuredByMeasType("UNKNOWN");
+		fetchedResource.setPolicyId(policyId);
+		fetchedResource.setProbableYield(11.22);
+		fetchedResource.setProductionGuarantee(33.44);
+		fetchedResource.setProductStatusCode("FINAL");
+
+		fetchedResource.setDataSyncTransDate(addSeconds(transactionDate, +1));
+		fetchedResource.setTransactionType(PoliciesSyncEventTypes.ProductUpdated);
+
+		service.synchronizeProduct(fetchedResource);
+		
+		ProductRsrc updatedResource = service.getProduct(topLevelEndpoints, productId.toString());
+
+		Assert.assertEquals(fetchedResource.getCommodityCoverageCode(), updatedResource.getCommodityCoverageCode());
+		Assert.assertEquals(fetchedResource.getCropCommodityId(), updatedResource.getCropCommodityId());
+		Assert.assertEquals(fetchedResource.getDeductibleLevel(), updatedResource.getDeductibleLevel());
+		Assert.assertEquals(fetchedResource.getInsuredByMeasType(), updatedResource.getInsuredByMeasType());
+		Assert.assertEquals(fetchedResource.getPolicyId(), updatedResource.getPolicyId());
+		Assert.assertEquals(fetchedResource.getProbableYield(), updatedResource.getProbableYield());
+		Assert.assertEquals(fetchedResource.getProductId(), updatedResource.getProductId());
+		Assert.assertEquals(fetchedResource.getProductionGuarantee(), updatedResource.getProductionGuarantee());
+		Assert.assertEquals(fetchedResource.getProductStatusCode(), updatedResource.getProductStatusCode());
+		
+		Assert.assertTrue(fetchedResource.getDataSyncTransDate().compareTo(updatedResource.getDataSyncTransDate()) == 0);
+		
+		//DELETE PRODUCT
+		updatedResource.setDataSyncTransDate(addSeconds(transactionDate, +1));
+		updatedResource.setTransactionType(PoliciesSyncEventTypes.ProductDeleted);
+		
+		service.synchronizeProduct(updatedResource);
+
+		ProductRsrc deletedResource = service.getProduct(topLevelEndpoints, productId.toString());
+		Assert.assertNull(deletedResource);
+
+		//CREATE Product
+		ProductRsrc resource2 = new ProductRsrc();
+		
+		resource2.setCommodityCoverageCode("CQG");
+		resource2.setCropCommodityId(16);
+		resource2.setDeductibleLevel(20);
+		resource2.setInsuredByMeasType("ACRES");
+		resource2.setPolicyId(policyId);
+		resource2.setProbableYield(12.34);
+		resource2.setProductId(productId);
+		resource2.setProductionGuarantee(56.78);
+		resource2.setProductStatusCode("OFFER");
+
+		resource2.setDataSyncTransDate(createTransactionDate);
+		resource2.setTransactionType(PoliciesSyncEventTypes.ProductCreated);
+
+		service.synchronizeProduct(resource2);
+		
+		ProductRsrc fetchedResource2 = service.getProduct(topLevelEndpoints, productId.toString()); 
+
+		Assert.assertEquals(resource2.getProductId(), fetchedResource2.getProductId());
+
+		//DELETE PRODUCT using the other delete method.
+		service.deleteProduct(topLevelEndpoints, productId.toString());
+
+		ProductRsrc deletedResource2 = service.getProduct(topLevelEndpoints, productId.toString());
```

Comment:
Shouldn't this return a 404 not found error because the resource doesn't exist anymore?